### ISSUE TITLE
wildcard ANAMEs fail validation

### ISSUE THREAD

**Issue by thinkl33t** [pwaring/dns-sync-zone#40]
The following aname record should be valid:

```
*.dev                   600 ANAME        docker-dev.colo
```

However dns-zone-sync fails it during validation:

```
 Error: Record failed validation in *.dev 600 ANAME docker-dev.colo
```

Adding it manually in the mythic beasts dns page appears to work:

![Screenshot from 2020-04-23 12-16-27](https://user-images.githubusercontent.com/1016780/80093429-5e12d780-855c-11ea-8164-2417392aeb1b.png)

**Comment by pwaring** [comment]
I've confirmed this with my `dnstest.uk` zone, I get the same validation error. The record is being tokenised correctly (`['*.dev', '300', 'ANAME', 'docker-dev.colo']`) so I think the cause of the problem is within `zone_validate.validate_tokens`. Will have a closer look now to see if this can be tweaked without breaking other record types.

**Comment by pwaring** [comment]
I'm pretty sure the problem lies in the `zone_validate.is_valid_name(record_hostname, strict_name)` function, which is being called as `zone_validate.is_valid_name('*.dev', True)`:

```python
def is_valid_name(name, strict=True):
    """Check validity of a hostname"""
    if name in ["@", "*"]:
        # Special cases
        return True
    return is_valid_domain(name, strict=strict)
```

That is handling the cases of origin and a wildcard on origin, and passing anything else down to `is_valid_domain`. I think what needs to happen is for `is_valid_name` to detect `*.` at the start of a domain, remove that and then call `is_valid_domain` with the remainder of the `record_hostname`. I need to check how wildcards work in DNS before knowing if that is the correct answer though.

**Comment by pwaring** [comment]
Checked understanding of wildcards, for sensible setups the only valid place to have `*.` is as the left-most token of the name (some DNS servers seem to support wildcards elsewhere, they will be considered broken for the purposes of this project).

Testing a change now, just waiting for Mythic to pick up the A/AAAA records associated with an ANAME.

**Comment by pwaring** [comment]
Okay, change made and tested. Please can you try the `paul/wildcard-validation` branch and let me know if it fixes the issue for you as well? It lops off the left-most wildcard (if one exists) and then validates the rest of the name as normal. I have tested with an ANAME that points to the same zone and one which points to an external zone, both seem to work.

**Comment by thinkl33t** [comment]
I can confirm the wildcard-validation branch works fine

**Comment by pwaring** [comment]
Thanks, I'll try and get a release out later today, or failing that tomorrow.

**Comment by pwaring** [comment]
Should be fixed in 1.3.0, please let me know if there are any problems.
